src/devices/machine/spi_sdcard.h

```diff
@@ -38,21 +38,36 @@ class spi_sdcard_device : public device_t
 	int m_type;
 
 private:
-	enum
+	enum sd_state
 	{
+		//REF Table 4-1:Overview of Card States vs. Operation Mode
 		SD_STATE_IDLE = 0,
+		SD_STATE_READY,
+		SD_STATE_IDENT,
+		SD_STATE_STBY,
+		SD_STATE_TRAN,
+		SD_STATE_DATA,
+		SD_STATE_DATA_MULTI, // synthetical state for this implementation
+		SD_STATE_RCV,
+		SD_STATE_PRG,
+		SD_STATE_DIS,
+		SD_STATE_INA,
+
+		//FIXME Existing states wich must be revisited
 		SD_STATE_WRITE_WAITFE,
 		SD_STATE_WRITE_DATA
 	};
```

Comment:
You need to specify an explicitly sized storage class for the enum if you want to register it for save states.